internal/utils/utils.go

```diff
@@ -21,3 +21,7 @@ func NewConversationID() string {
 	}
 	return hex.EncodeToString(buf)
 }
+
+func Ptr[T any](v T) *T {
+	return &v
+}
```

Comment:
I remember reading Go added a language feature so that you don't have to do this anymore. But it's only coming in Go 1.26 - https://github.com/golang/go/issues/45624